crates/accelerate/src/euler_one_qubit_decomposer.rs

```diff
@@ -853,6 +970,112 @@ pub fn params_zxz(unitary: PyReadonlyArray2<Complex64>) -> [f64; 4] {
     params_zxz_inner(mat)
 }
 
+type OptimizeDecompositionReturn = Option<((f64, usize), (f64, usize), OneQubitGateSequence)>;
+
+#[pyfunction]
+pub fn optimize_1q_gates_decomposition(
+    runs: Vec<Vec<PyRef<DAGOpNode>>>,
+    qubits: Vec<usize>,
+    bases: Vec<Vec<PyBackedStr>>,
+    simplify: bool,
+    error_map: Option<&OneQubitGateErrorMap>,
+    atol: Option<f64>,
+) -> Vec<OptimizeDecompositionReturn> {
+    runs.iter()
+        .enumerate()
+        .map(|(index, raw_run)| -> OptimizeDecompositionReturn {
+            let mut error = match error_map {
+                Some(_) => 1.,
+                None => raw_run.len() as f64,
+            };
+            let qubit = qubits[index];
+            let operator = &raw_run
+                .iter()
+                .map(|node| {
+                    if let Some(err_map) = error_map {
+                        error *=
+                            compute_error_term(node.instruction.operation.name(), err_map, qubit)
+                    }
+                    node.instruction
+                        .operation
+                        .matrix(&node.instruction.params)
+                        .unwrap()
+                })
+                .fold(
+                    [
+                        [Complex64::new(1., 0.), Complex64::new(0., 0.)],
+                        [Complex64::new(0., 0.), Complex64::new(1., 0.)],
+                    ],
+                    |mut operator, node| {
+                        matmul_1q(&mut operator, node);
+                        operator
+                    },
+                );
+            let old_error = if error_map.is_some() {
+                (1. - error, raw_run.len())
+            } else {
+                (error, raw_run.len())
+            };
+            let target_basis_list = &bases[index];
+            let mut target_basis_vec: Vec<EulerBasis> = Vec::with_capacity(target_basis_list.len());
+            for basis in target_basis_list {
+                let basis_enum = EulerBasis::__new__(basis.deref()).unwrap();
+                target_basis_vec.push(basis_enum)
+            }
```

Comment:
```suggestion
            let target_basis_vec: Vec<EulerBasis> = bases[index]
                .iter()
                .map(|basis| EulerBasis::__new__(basis).unwrap())
                .collect();
```

Reply:
I applied this suggestion to the branch in: https://github.com/Qiskit/qiskit/pull/12650/commits/68af440ca1ad08d3a0551c7e00013eddb29b2434